### PR TITLE
Feat - Add Function Name Tags

### PR DESCRIPTION
**Pull request by PineappleIOnic**
## What does this PR do?
This PR Introduces the ability to name function tags to be more easily identifiable.

## Test Plan

I have added a new name field to tags, I have also implemented the relevant tests to ensure this is working as expected.

`POST v1/functions/:functionID/tags` has been updated to optionally accept a `name` field. If no name is given a blank string is used instead.

## Related PRs and Issues
https://github.com/appwrite/appwrite/issues/1222

### Have you read the [Contributing Guidelines on issues](https://github.com/appwrite/appwrite/blob/master/CONTRIBUTING.md)?
Yes

### CODE REVIEW

File: app/controllers/api/functions.php [hunk 1]
```diff
@@ -445,11 +445,12 @@
     ->param('functionId', '', new UID(), 'Function unique ID.')
     ->param('command', '', new Text('1028'), 'Code execution command.')
     ->param('code', [], new File(), 'Gzip file with your code package. When used with the Appwrite CLI, pass the path to your code directory, and the CLI will automatically package your code. Use a path that is within the current directory.', false)
+    ->param('name', '', new Text(512), 'Function Name', true)
```
Comment (eldadfux) [inline app/controllers/api/functions.php:448]: I would probably have the name as the first param here.

File: src/Appwrite/Utopia/Response/Model/Tag.php [hunk 2]
```diff
@@ -34,6 +34,12 @@ public function __construct()
                 'default' => '',
                 'example' => 'enabled',
             ])
+            ->addRule('name', [
```
Comment (eldadfux) [inline src/Appwrite/Utopia/Response/Model/Tag.php:37]: I would move the name above `command`

File: src/Appwrite/Utopia/Response/Model/Tag.php [hunk 3]
```diff
@@ -34,6 +34,12 @@ public function __construct()
...
+                'type' => self::TYPE_STRING,
+                'description' => 'Function Name.',
+                'default' => '',
+                'example' => 'Hello World Script',
```
Comment (eldadfux) [inline src/Appwrite/Utopia/Response/Model/Tag.php:41]:
```suggestion
                'example' => 'Hello World Tag',
```

**Review by eldadfux — CHANGES_REQUESTED** [review]

**Comment by eldadfux** [comment]
Waiting for progress on the [Custom IDs](https://github.com/appwrite/rfc/pull/26) RFC which might affect this breaking change.

**Comment by PineappleIOnic** [comment]
Is this still a active PR or should I close it? @eldadfux

**Pull request closed without merging**